swift/README.md

```diff
@@ -369,25 +369,25 @@ Generic functions:
 https://developer.apple.com/library/mac/documentation/Swift/Conceptual/Swift_Programming_Language/Generics.html#//apple_ref/doc/uid/TP40014097-CH26-ID181
 
 ```swift
-func getValue<T>(dictionary: [String: AnyObject], key: String, errorHandler: String -> Void)) -> T? {
-    if let value = dictionary[key] {
-        if let value = value as? T {
-            return value
-        } else {
-            errorHandler("\(key) is not a \(T.self)")
-        }
-    } else {
+func getValue<T>(dictionary: [String: AnyObject], key: String, errorHandler: String -> Void) -> T? {
+    guard let value = dictionary[key] else {
         errorHandler("\(key) is not present")
+        return nil
+    }
+
+    guard let rightType = value as? T else {
+        errorHandler("\(key) is not a \(T.self)")
+        return nil
     }
 
-    return nil
+    return rightType
 }
 ```
 
-Note that the type T is inferred by what you assign `getValue()` to.  So in this case getValue() infers String? because `message` is a String?.
+Note that the type `T` is inferred by what you assign `getValue()` to.  So in this case getValue() infers `String?` because `message` is a `String?`.
```

Comment:
@coffee-cup Back in the day that's how keyboarding was taught (two spaces after a period).  You can still find holy wars about that on the internet. :)